pennylane_lightning_gpu/lightning_gpu.py

```diff
@@ -152,9 +233,126 @@ def syncH2D(self, use_async=False):
 
     def syncD2H(self, use_async=False):
         """Explicitly synchronize GPU data to CPU"""
+        if len(self._state) != 2**self.num_wires:
+            state = np.zeros(2**self.num_wires, dtype=np.complex128)
+            state = self._asarray(state, dtype=self.C_DTYPE)
+            self._state = self._reshape(state, [2] * self.num_wires)
+
         self._gpu_state.DeviceToHost(self._state.ravel(order="C"), use_async)
         self._pre_rotated_state = self._state
 
+    @property
+    def state(self):
+        # Flattening the state.
+        shape = (1 << self.num_wires,)
+        self.syncD2H()
+        return self._reshape(self._pre_rotated_state, shape)
+
+    def _create_basis_state_default(self, index=0):
+        """Return a default state to construct the self._state with a size of [2] on the host.
+        [2] is chose to reduce the overhead of memory allocation on the host. Once all of data of
+        self._state is needed, self._state will be copied from the GPU and memory allocation of
+        full scale for self._state will be conducted in the method 'syncD2H()'.
+        Args:
+            index (int): integer representing the computational basis state
+        Returns:
+            array[complex]: complex array of shape ``[2]``
+        """
+
+        state = np.zeros(2, dtype=np.complex128)
+        state[index] = 1
+        state = self._asarray(state, dtype=self.C_DTYPE)
+        return self._reshape(state, [2])
```

Comment:
Why do we need to create a "fake state", before returning the true state from the GPU?
Also, why size 2?

Reply:
just want to keep the `_state` attribute here to make the testing simpler. But need further discussions. We can set it as an empty np.array as well.

Reply:
removed!